contracts/examples/DXswapSwapRelayer.sol

```diff
@@ -0,0 +1,296 @@
+
+contract DXswapSwapRelayer {
+    using SafeMath for uint256;
+
+    event NewOrder(uint256 indexed _orderIndex);
+
+    event ExecutedOrder(uint256 indexed _orderIndex);
+
+    event WithdrawnExpiredOrder(uint256 indexed _orderIndex);
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    event Details(uint256 amountIn, uint256 amountOut);
+
+    struct Order {
+        address tokenIn;
+        address tokenOut;
+        uint256 amountIn;
+        uint256 priceTolerance;
+        uint256 minReserveA;
+        uint256 minReserveB;
+        address oraclePair;
+        uint256 deadline;
+        uint256 maxWindowTime;
+        uint256 oracleId;
+        address factory;
+        bool executed;
+    }
+
+    uint256 public immutable GAS_ORACLE_UPDATE = 168364;
+    uint256 public immutable PARTS_PER_MILLION = 1000000;
+    uint256 public immutable BOUNTY = 0.01 ether; // To be decided
+    uint8 public immutable PROVISION = 1;
+    uint8 public immutable REMOVAL = 2;
+
+    address public immutable dxSwapFactory;
+    address public immutable dxSwapRouter;
+    address public immutable uniswapFactory;
+    address public immutable uniswapRouter;
+    address public immutable WETH;
+
+    OracleCreator oracleCreator;
+    uint256 public orderCount;
+    mapping(uint256 => Order) orders;
+    address payable public owner;
+
+    constructor(
+        address payable _owner,
+        address _dxSwapFactory,
+        address _dxSwapRouter,
+        address _uniswapFactory,
+        address _uniswapRouter,
+        address _WETH,
+        OracleCreator _oracleCreator
+    ) public {
+        owner = _owner;
+        dxSwapFactory = _dxSwapFactory;
+        dxSwapRouter = _dxSwapRouter;
+        uniswapFactory = _uniswapFactory;
+        uniswapRouter = _uniswapRouter;
+        WETH = _WETH;
+        oracleCreator = _oracleCreator;
+    }
+
+    function createSwapOrder(
+        address tokenA,
+        address tokenB,
+        uint256 amountInTokenA,
+        uint256 amountInTokenB,
+        uint256 priceTolerance,
+        uint256 minReserveA,
+        uint256 minReserveB,
+        uint256 maxWindowTime,
+        uint256 deadline,
+        address factory
+    ) external payable returns (uint256 orderIndex) {
+        require(factory == dxSwapFactory || factory == uniswapFactory, 'DXswapRelayer: INVALID_FACTORY');
+        require(msg.sender == owner, 'DXswapRelayer: CALLER_NOT_OWNER');
+        require(tokenA != tokenB, 'DXswapRelayer: INVALID_PAIR');
+        require(tokenA < tokenB, 'DXswapRelayer: INVALID_TOKEN_ORDER');
+        // only one token amount can be > 0 and second token amount is calculated as output
+        require(
+            (amountInTokenA > 0 && amountInTokenB == 0) || (amountInTokenA == 0 && amountInTokenB > 0),
+            'DXswapRelayer: INVALID_TOKEN_AMOUNT'
+        );
+        require(priceTolerance <= PARTS_PER_MILLION, 'DXswapRelayer: INVALID_TOLERANCE');
```

Comment:
to consult with the client: it may require that priceTolerance is at least 0.3% as this the fee uniswap charges.
However, it dxswap has different fee model, than perhaps it's better to leave as is